src/core/Track.cpp

```diff
@@ -1070,12 +1112,56 @@ void TrackContentObjectView::mouseReleaseEvent( QMouseEvent * me )
 	{
 		setSelected( !isSelected() );
 	}
-
-	if( m_action == Move || m_action == Resize || m_action == ResizeLeft )
+	else if( m_action == Move || m_action == Resize || m_action == ResizeLeft )
 	{
 		// TODO: Fix m_tco->setJournalling() consistency
 		m_tco->setJournalling( true );
 	}
+	else if( m_action == Split )
+	{
+		SampleTCO * leftTCO = dynamic_cast<SampleTCO*>( m_tco );
+
+		leftTCO->setMarkerEnabled( false );
+
+		const int relativePixelPos = me->pos().x();
+		const float ppt = m_trackView->trackContainerView()->pixelsPerTact();
+		MidiTime splitPos = relativePixelPos * MidiTime::ticksPerTact() / ppt;
+		const float snapSize = gui->songEditor()->m_editor->getSnapSize();
+
+		//Essentially duplicate logic from knifeMarkerPos, but shift needs to be inverted for some reason
+		if ( me->modifiers() & Qt::ControlModifier
+		  || me->modifiers() & Qt::AltModifier    ){}
+		else if ( me->modifiers() & Qt::ShiftModifier )
+		{	//If shift is held we quantize the length of the new left clip...
+			const MidiTime leftPos = splitPos.quantize( snapSize );
+			//...or right clip...
+			const MidiTime rightOff = m_tco->length() - splitPos;
+			const MidiTime rightPos = m_tco->length() - rightOff.quantize( snapSize );
+			//...whichever gives a position closer to the cursor
+			if ( abs(leftPos - splitPos) < abs(rightPos - splitPos) ) splitPos = leftPos;
+			else splitPos = rightPos;
+		}
+		else
+		{
+			splitPos = MidiTime(splitPos + m_initialTCOPos).quantize( snapSize ) - m_initialTCOPos;
+		}
+		splitPos += m_initialTCOPos;
+
+		//Don't do anything if we slid off the TCO
+		if ( splitPos <= m_initialTCOPos || splitPos >= m_initialTCOEnd ){}
```

Comment:
Why you don't ask `if (splitPos >= m_initialTCOPos || splitPos <= m_initialTCOEnd)` ?